### ISSUE TITLE
[question] how to address the case where different models in a project output variable number of outputs

### ISSUE THREAD

**Issue by walkacross** [pytorch/ignite#155]
@alykhantejani @jasonkriss @vfdev-5
Hey, thanks for posting this great framework for training pytorch model.

If a project need to train two models but with variable number of outputs (like y_pred, other_output = model(x)), do I have to create two self-defined create_supervised_trainer?  any suggestion for this case?

Have a good day.

**Comment by vfdev-5** [comment]
@walkacross actually `create_supervised_trainer` is a helper method to create a trainer, feel free to copy and edit it locally in your application to satisfy your needs.
If your model outputs like this `y_pred, other_output = model(x)` you can have two possibilities:
- use `create_supervised_trainer` and adapt loss function to handle this particular output
- use a custom trainer and filter out the output to feed in the appropriate `y` to the standard loss functions

HTH

**Comment by walkacross** [comment]
@vfdev-5  get it and thanks for your kind reply.

**Comment by vfdev-5** [comment]
@walkacross feel free to ask and take a look at not yet released documentation: [concepts](https://github.com/pytorch/ignite/blob/master/docs/source/concepts.rst), [quickstart](https://github.com/pytorch/ignite/blob/master/docs/source/quickstart.rst) and [examples](https://github.com/pytorch/ignite/tree/master/examples).
Here I made some [more examples](https://github.com/vfdev-5/ignite-examples).
